### DISCUSSION TITLE
How to remove Remote Sensors

### DISCUSSION THREAD

**Discussion by 42Network** [PeteRager/lennoxs30#252] [Q&A]
My installer added three Lennox BLE Remote Sensors to my S40 for a few weeks, which HA discovered, but now the sensors are removed and returned to the installer as they did not solve my particular problem. The three sensors are deleted from my S40 thermostat but still appear in HA. What is the right way to fully remove them from HA?

Thanks

**Comment by PeteRager** [comment]
Do the sensor have values or are they marked as Unknown?

If unknown - this means that the integration no longer provides the sensors - which is what I'd expect. In this case go into HASS settings, go to entities and remove these entities.  There should be an icon that shows the unknown ones and you can sort of that column.

If they are still reporting values, this means they really have not been removed from the S40 configuration.  In this case go into settings integrations and click off the devices link for the integration, find the entities and disable them.

**Reply by 42Network** [reply]
They all say "entity unavailable" but the Delete button at bottom left in the details dialog box is greyed out.

**Reply by 42Network** [reply]
To answer my own question or add to the question, when the installer added the BLE sensors he disabled my zone system. Once he removed the BLE sensors and re-enabled the zone system, HA was confused somehow and now my zone system entities are all prefixed with "s40_3_zone_*" whereas all my other entities are prefixed with "s40_*" (i.e. without the _3).

Do you know where this extra "_3" came from? Perhaps I should delete the entire S30/S40 integration and start over? If I were to do so, is there anyway to preserve any of my historical data? Or will my history only go back to the time of reinstall of S30/40 integration? Do you have instructions for reinstall?

Thanks/Nathan

**Comment by PeteRager** [comment]
The prefix comes from the Name the S40 has.  So in your system the Name seems to be set to S40 (and for example I have mine set to RageHouse).  You may want to check to make sure the name wasn't changed to "S40_3" in the panel.  

The pattens for entityid is climate.<Name>_zone_<zonename>

If an integration tries to create an entity with an Id that already exists HASS will append an _1 at the end, but since you see the 3 in the middle, I don't think that is happening.  

The recorder data is tied to the entity id, so you can delete the integration and readd it and you SHOULD not lose the historical data.

If the name hasn't changed, we could enable debug logging which would show us exactly why the entity's are being created this way.

**Comment by PeteRager** [comment]
If not done, you may want to do a full restart of HASS, that may sort some of the issues out.

Since you are not able to delete the BLE sensors, that means that the integration is registering them at startup, which means that while they are removed from the system they are not removed from the configuration data the S40 sends.  This config is gotten from the S40 on every startup.  It's also possible that the integration is not interpreting this data correctly when the devices have been removed.  Capturing a message log would allow us to see if that is the case or not.

**Reply by 42Network** [reply]
I have done a couple of full HA restarts as well as Lennox system restarts. I bet the _3 comes from some previous history where it is possible I "rediscovered" my S40 more than twice. And yeah, my system name is generic: it is simply "S40" and that is is what the thermostat itself thinks its name is, and most of the entities start with s40_ now, except for everything related to the zoning, which start with s40_3_.


As for historical metrics, I guess I don't mind starting over, since everything is additionally stored in InfluxDB in my case, so with some tweaking in Grafana I can still see my old data to compare with new data.

So I can enable debugging and logs for you if you want to perhaps troubleshoot this a bit more to make the code more robust for things like upgrading from S30 to S40 as I have done, as well as adding & removing BLE sensors.

Or we can throw up our hands and I start over from scratch, if you can tell me how to make sure there are no traces of S40 or S40_3 entities in my HA before reinstalling the Lennox integration.

Thanks/Nathan

**Reply by PeteRager** [reply]
Apologies for the delay.

I think a good first step is to have a documented process for those upgrade steps - even if it involves removing the integration and readding.  Detecting and managing an upgrade from s30 to s40 automatically will be challenging.

The history is tied to the entitiy_id, removing an entity_id does not remove the history.

Internally all the entities have a unique Id. The unique Id prefix is the serial number of the S30/S40 controller. Since these will be different I would expect duplicates entities to appear with _1. But I don't have any explanation as to why we'd see _3 and it may indicate a deeper issue.  So if you want to spend the time getting the needed logs, I'd be happy to look at it as I don't like mysteries as they tend to eventually crop out and cause major issues.

I do think the best approach to get your system healthy (after capturing the logs) is to remove the integration.  The integration has a unique id and all the entities in the .storage/config.entity_registry are attached to this unique Id. Hence when the integration is removed HASS removes the related entities from this file.  But I don't think this will solve the BLE sensor issue, so let's wait on this until we've looked at the log files.

We will need two sets of logs, the first one is a message log - this will contain all the S40 config messages, I have a simulator I can feed this into and replicate your setup. The second is the debug information in the home-assistant.log, this will show us all the entities being created and their names. With that we should get a complete view of why the BLE sensors are being created and if the integration is putting the _3 in or not.

Instructions are here:
https://github.com/PeteRager/lennoxs30#reporting-bugs

1. Add the debug stuff to configuration.yaml
2. In the integration config check the "Log S30 messages to a file", but do not check "Debug logging logs messages", give it a file name
3. Restart HASS
4. Grab the home-assistant.log file and the message log file 
5. Post here or I can give you an email to send to.

**Comment by 42Network** [comment]
Attached is my home-assistant.log after enabling Lennox debugging. I configured a separate messages log called nathan_s40.log, but it is still empty after a restart.
[home-assistant.log](https://github.com/PeteRager/lennoxs30/files/12251765/home-assistant.log)

**Comment by PeteRager** [comment]
Thanks, that log file had what we needed,

- There is no evidence that the Ble sensors are being created, so they have been removed from the Lennox device and the integration is not trying to add them
- There are no _3 being added to the climate entities

Net is the integration appears to be working properly. Now I don't know why HASS is adding the _3 or why it won't allow you to remove the BLE sensors.  It should be easy for me to simulate the Ble sensors being removed.  I'll try that over the next couple of days.

Here is a process that should get your system cleaned up; or you can wait for me to simulate the removal.

1. Backup HASS
2. Delete the integration
3. Restart HASS
4. Go into settings / entities and verify all the entities related to the integration have been removed. HASS should delete them all when the owning integration is removed. If that is not the case let me know.
5. Readd the integration, entities should get recreated with the same entity ids and your history should still be there.

**Reply by 42Network** [reply]
I can wait for your simulation, no problem. Thanks.

**Reply by 42Network** [reply]
Any updates on removal? I have not yet done your clean reinstall steps and my HA still complains about the remote sensors not sending data. 

Thanks.

**Reply by 42Network** [reply]
To complete this thread, I ended up removing the integration, upgraded the firmware on my S40, and reinstalled the integration. As a result, I no longer see any references to the old remote sensors in HA. I can only conclude something inside HA or the integration was persisting the remote sensors even though they were removed from the Lennox system long ago.
